fv3core/stencils/map_single.py

```diff
@@ -181,30 +75,49 @@ class MapSingle:
     """
 
     def __init__(self, kord: int, mode: int, i1: int, i2: int, j1: int, j2: int):
-        self._grid = spec.grid
-        shape = self._grid.domain_shape_full(add=(1, 1, 1))
-        origin = self._grid.compute_origin()
+        shape = spec.grid.domain_shape_full(add=(1, 1, 1))
+        origin = spec.grid.compute_origin()
 
         self.dp1 = utils.make_storage_from_shape(shape, origin=origin)
         self.q4_1 = utils.make_storage_from_shape(shape, origin=origin)
         self.q4_2 = utils.make_storage_from_shape(shape, origin=origin)
         self.q4_3 = utils.make_storage_from_shape(shape, origin=origin)
         self.q4_4 = utils.make_storage_from_shape(shape, origin=origin)
 
-        self.i_extent = i2 - i1 + 1
-        self.j_extent = j2 - j1 + 1
+        self.lev = utils.make_storage_from_shape(
```

Comment:
Temporary fields do not need to be cached in OOP-ified stencils.

Reply:
But we don't want this re-allocated on each call, right? I'm not opposed to switching it to a function-local temporary, and I suppose right now everything is cached anyway, but this might not be how it works in the future. I'll leave the changes to others who are more familiar with fv3core's up-to-date standards.